sw/device/lib/dif/dif_sysrst_ctrl.h

```diff
@@ -17,7 +17,377 @@
 extern "C" {
 #endif  // __cplusplus
 
-// TODO: Add DIFs.
+/**
+ * A System Reset Controller key combination.
+ */
+typedef enum dif_sysrst_ctrl_key_combo {
+  /**
+   * Key combination 0.
+   */
+  kDifSysrstCtrlKeyCombo0 = 1U << 0,
+  /**
+   * Key combination 1.
+   */
+  kDifSysrstCtrlKeyCombo1 = 1U << 1,
+  /**
+   * Key combination 2.
+   */
+  kDifSysrstCtrlKeyCombo2 = 1U << 2,
+  /**
+   * Key combination 3.
+   */
+  kDifSysrstCtrlKeyCombo3 = 1U << 3,
+  /**
+   * All key combination ORed together.
+   *
+   * This is useful when clearing all key combination IRQ causes at once, e.g.,
+   * when initializing the System Reset Controller.
+   */
+  kDifSysrstCtrlKeyComboAll = (1U << 4) - 1,
+} dif_sysrst_ctrl_key_combo_t;
+
+/**
+ * System Reset Controller keys that can form a key combination.
+ */
+typedef enum dif_sysrst_ctrl_key {
+  /**
+   * Key 0.
+   */
+  kDifSysrstCtrlKey0 = 1U << 0,
+  /**
+   * Key 1.
+   */
+  kDifSysrstCtrlKey1 = 1U << 1,
+  /**
+   * Key 2.
+   */
+  kDifSysrstCtrlKey2 = 1U << 2,
+  /**
+   * Power button key.
+   */
+  kDifSysrstCtrlKeyPowerButton = 1U << 3,
+  /**
+   * AC power preset key.
+   */
+  kDifSysrstCtrlKeyAcPowerPresent = 1U << 4,
+} dif_sysrst_ctrl_key_t;
+
+/**
+ * System Reset Controller key combination detection actions.
+ */
+typedef enum dif_sysrst_ctrl_key_combo_action {
+  /**
+   * Disable / disconnect battery action.
+   */
+  kDifSysrstCtrlKeyComboActionBatterDisable = 1U << 0,
+  /**
+   * Issue an interrupt to the processor action.
+   */
+  kDifSysrstCtrlKeyComboActionInterrupt = 1U << 1,
+  /**
+   * Assert the embedded controller reset for a specified duration action.
+   */
+  kDifSysrstCtrlKeyComboActionEcReset = 1U << 2,
+  /**
+   * Issue a POR reset request to the reset manager block action.
+   *
+   * TODO: confirm this.
+   */
+  kDifSysrstCtrlKeyComboActionSelfReset = 1U << 3,
+} dif_sysrst_ctrl_key_combo_action_t;
+
+/**
+ * Runtime configuration for the System Reset Controller key combination
+ * detection feature.
+ */
+typedef struct dif_sysrst_ctrl_key_combo_config {
+  /**
+   * A key combination to configure.
+   */
+  dif_sysrst_ctrl_key_combo_t key_combo;
+  /**
+   * The keys that comprise the key combination to detect (i.e., one or more
+   * `dif_sysrst_ctrl_key_t`s ORed together).
+   */
+  uint32_t keys;
```

Comment:
Enums for flags are ever-present in our code base, and just be aware that where they work well in C, they can be awkward and full of casts in C++.